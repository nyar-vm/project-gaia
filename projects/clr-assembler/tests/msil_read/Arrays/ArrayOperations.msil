.class public auto ansi beforefieldinit
    Valkyrie.Translator.Arrays.ArrayOperations
        extends [System.Runtime]System.Object
{

    .method public hidebysig static void
        Main() cil managed
    {
        .maxstack 8
        .locals init (
            [0] int32[] singleArray,
            [1] int32[0...,0...] multiArray,
            [2] int32[][] jaggedArray,
            [3] int32 length,
            [4] string[] stringArray,
            [5] int32[] copyArray
        )

        // singleArray = new int32[5]
        IL_0000: ldc.i4.5
        IL_0001: newarr       [System.Runtime]System.Int32
        IL_0006: stloc.0      // singleArray

        // singleArray[0] = 10
        IL_0007: ldloc.0      // singleArray
        IL_0008: ldc.i4.0
        IL_0009: ldc.i4.s     10
        IL_000b: stelem.i4

        // singleArray[1] = 20
        IL_000c: ldloc.0      // singleArray
        IL_000d: ldc.i4.1
        IL_000e: ldc.i4.s     20
        IL_0010: stelem.i4

        // multiArray = new int32[3, 4]
        IL_0011: ldc.i4.3
        IL_0012: ldc.i4.4
        IL_0013: newobj       instance void int32[0...,0...]::.ctor(int32, int32)
        IL_0018: stloc.1      // multiArray

        // multiArray[0, 0] = 1
        IL_0019: ldloc.1      // multiArray
        IL_001a: ldc.i4.0
        IL_001b: ldc.i4.0
        IL_001c: ldc.i4.1
        IL_001d: call         instance void int32[0...,0...]::Set(int32, int32, int32)

        // jaggedArray = new int32[3][]
        IL_0022: ldc.i4.3
        IL_0023: newarr       int32[]
        IL_0028: stloc.2      // jaggedArray

        // jaggedArray[0] = new int32[4]
        IL_0029: ldloc.2      // jaggedArray
        IL_002a: ldc.i4.0
        IL_002b: ldc.i4.4
        IL_002c: newarr       [System.Runtime]System.Int32
        IL_0031: stelem.ref

        // length = singleArray.Length
        IL_0032: ldloc.0      // singleArray
        IL_0033: ldlen
        IL_0034: conv.i4
        IL_0035: stloc.3      // length

        // stringArray = new string[] { "Hello", "World", "Array" }
        IL_0036: ldc.i4.3
        IL_0037: newarr       [System.Runtime]System.String
        IL_003c: dup
        IL_003d: ldc.i4.0
        IL_003e: ldstr        "Hello"
        IL_0043: stelem.ref
        IL_0044: dup
        IL_0045: ldc.i4.1
        IL_0046: ldstr        "World"
        IL_004b: stelem.ref
        IL_004c: dup
        IL_004d: ldc.i4.2
        IL_004e: ldstr        "Array"
        IL_0053: stelem.ref
        IL_0054: stloc.s      stringArray

        // copyArray = new int32[singleArray.Length]
        IL_0056: ldloc.0      // singleArray
        IL_0057: ldlen
        IL_0058: conv.i4
        IL_0059: newarr       [System.Runtime]System.Int32
        IL_005e: stloc.s      copyArray

        IL_0060: ret

    } // end of method ArrayOperations::Main

    .method public hidebysig specialname rtspecialname instance void
        .ctor() cil managed
    {
        .maxstack 8

        IL_0000: ldarg.0      // this
        IL_0001: call         instance void [System.Runtime]System.Object::.ctor()
        IL_0006: ret

    } // end of method ArrayOperations::.ctor

} // end of class Valkyrie.Translator.Arrays.ArrayOperations