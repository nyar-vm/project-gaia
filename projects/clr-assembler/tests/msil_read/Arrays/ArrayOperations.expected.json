{
    "assembly_name": null,
    "extern_assemblies": [],
    "module_name": null,
    "class_name": "Valkyrie.Translator.Arrays.ArrayOperations",
    "class_modifiers": [
        "public",
        "auto",
        "ansi",
        "beforefieldinit"
    ],
    "extends": "[System.Runtime]System.Object",
    "method_count": 2,
    "method_names": [
        "Main",
        ".ctor"
    ],
    "statement_count": 1
}